### PR TITLE
Iron180

### PR DESCRIPTION
**Pull request by Gondolindrim**
Adds support for the Iron180 keyboard.

## Description

Adds support for the Iron180 keyboard.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/iron180/config.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x8384 //SR for Smith and Rune
+#define PRODUCT_ID      0x1180
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    SmithRune
+#define PRODUCT         IRON180
+#define DESCRIPTION	Smith and Rune Iron180
```
Comment (drashna) [inline keyboards/iron180/config.h:26]:
```suggestion
```

File: keyboards/iron180/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#define ____ KC_NO
```
Comment (drashna) [inline keyboards/iron180/keymaps/default/keymap.c:17]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

Comment (drashna) [reply keyboards/iron180/keymaps/default/keymap.c:17]: Actually, I'm seeing this a lot, why is this here?  Did you use a site to generate the code?  If so, which?  

Or is it copying from another keyboard? 

Or?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ck60i/rules.mk [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality 
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BACKLIGHT_DRIVER = pwm
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/ck60i/rules.mk:20]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/iron180/rules.mk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/iron180/rules.mk:19]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by noroadsleft** [comment]
@Gondolindrim is the CK60i supposed to be included on this PR? You've got two keyboards here but your post only mentions one.

File: keyboards/ck60i/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#define ____ KC_NO
```
[the remark this answers is not in the document]
Comment (Gondolindrim) [reply keyboards/ck60i/keymaps/default/keymap.c:17]: Committed.

**Comment by Gondolindrim** [comment]
> @Gondolindrim is the CK60i supposed to be included on this PR? You've got two keyboards here but your post only mentions one.

I branched the iron180 branch from the ck60i branch instead of the master one; I will commit a deletion of those, but if a new PR is necessary please tell me

File: keyboards/iron180/keymaps/default/keymap.c (hunk 2 above)
Comment (Gondolindrim) [reply keyboards/iron180/keymaps/default/keymap.c:17]: I had no idea that there already was a quantum definition, hence why I was using that four underscores one. I've always done that, can't remember if it was an old habit or just copied over. I will try to use the seven underscores one from now on

Comment (Gondolindrim) [reply keyboards/iron180/keymaps/default/keymap.c:17]: Commited in 9f6d2dbea

File: keyboards/ck60i/keymaps/test/keymap.c [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#define ____ KC_NO
```
[the remark this answers is not in the document]
Comment (Gondolindrim) [reply keyboards/ck60i/keymaps/test/keymap.c]: Commited in e55cca22c

File: keyboards/iron180/rules.mk (hunk 4 above)
Comment (Gondolindrim) [reply keyboards/iron180/rules.mk:19]: Commited in 28c956bfe

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**